docs/DeveloperGuide.md

```diff
@@ -1,13 +1,413 @@
+It uses Boolean Methods to determine the presence of keywords, and then creates a Validator class to process the command
+should the keywords be present.
+
+The following are some examples:
+
+| Boolean Methods                        | Check if input starts with | Feature Requires | Creates                                            |
+|----------------------------------------|----------------------------|------------------|----------------------------------------------------|
+| ListMonthlyExpensesCommand.isCommand() | list monthly expenses      | input            | ListMonthlyExpensesValidator.processCommand(input) |
+| AddExpenseCommand.isCommand()          | add expense                | input            | AddExpenseValidator.processCommand(input)          | 
+
+The Parser class is also used to parse lines from the `Storage` .txt file and loads it into the main application upon
+restart. This will be covered more under the storage class.
+
+#### 3.3 UI Class
+
+The UI Class is to print out elements of the app in the CLI. It contains many methods used to print general, often-used
+messages such as displayWelcomeMessage() and displayExitMessage().
+
+#### 3.4 Command Class
+
+The `Command` class is a parent class which is responsible for checking for input keywords. It has multiple subclasses,
+which corresponds to a specific function of the application. Each of its subclass checks for specific keywords in the
+user input. If the input is valid as checked by the `isCommand()` method, it will call its corresponding validator.
+
+#### 3.5 Validator Classes
+
+The Validator classes are a group of classes which is responsible for further checks on input, such as whether it has
+the right formatting and contains all necessary details. It will return different subclasses of the command object
+depending on whether the input is valid or not. Once the object is instantiated and returned, the `main` class will call
+`execute` to execute the feature.
+
+##### 3.4.1 and 3.5.1 XML Sequence Diagram
+
+The following sequence diagram shows the process of what happens when a user input is passed through the application,
+until it gets executed.
+
+**Note**: 
+1. Validator class is not a parent class, but we have grouped it under validators to prevent cluttering of 
+the diagrams. 
+2. Validator also returns different type of Command Classes depending on the validity of input, which has been omitted
+for the same reason as above.
+
+![CommandClass.drawio.png](diagrams/CommandClass.drawio.png)
+
+#### 3.6 Expense and Income Class
+The `Expense` and `Income` class inherits from the Transaction class.
+
+`Expense` class stores one expense record given by the user.
+
+`Income` class stores one income record given by the user.
+
+The methods for `Expense` and `Income` are not shown.
+
+
+![ExpenseAndIncomeClassDiagram.drawio.png](diagrams/ExpenseAndIncomeClassDiagram.drawio.png)
```

Comment:
Should variables' accessibility (private public etc) be shown in the class diagram once it has been decided? (there is no default accessibility in class diagrams)
![image](https://github.com/user-attachments/assets/23a902d1-31b5-42fd-8ad3-cade5fd9ac7b)